### ISSUE TITLE
Typescript interface code for  GraphQL Type that implements two GraphQL interfaces is not generated correctly

### ISSUE THREAD

**Issue by pashazz** [dotansimha/graphql-code-generator#1577]
I have a Type that implements two Interfaces.
```graphql
type GVM implements GAbstractVM & GAclXenObject {
  #...
}
```
if I use a fragment that includes two fragments on these interfaces...
```graphql
fragment VMInfoFragment on GVM {
  ...ACLXenObjectFragment
  ...AbstractVMFragment
#...
}
```

Then a TS type is generated the following way:
```ts
export namespace VmInfoFragment {
  export type Fragment = {
//...
  } & (AclXenObjectFragment.Fragment | AbstractVmFragment.Fragment);
```

However it should be generated as:
```ts
export namespace VmInfoFragment {
  export type Fragment = {
//...
  } & (AclXenObjectFragment.Fragment & AbstractVmFragment.Fragment);
```

Can be reproduced in latest version 0.18.0

Minimal working example: https://github.com/pashazz/gqlgenmwe
Where it goes wrong: https://github.com/pashazz/gqlgenmwe/blob/eaa366a8081bcd36d7d50b81db5cb301bba24127/generated-models.ts#L53

**Comment by dotansimha** [comment]
I can confirm that this issue also happens on v1.0. Our current logic handles a case where fragments are in use on a simple type, interface or union.

Fixed in:  https://github.com/dotansimha/graphql-code-generator/pull/1580

**Comment by dotansimha** [comment]
Fixed in 1.0.5.
